tests/integration/conftest.py

```diff
@@ -218,63 +216,63 @@ async def app_fixture(
         f"'set -euo pipefail; echo \"{inline_yaml}\" | {pebble_exec} -- {discourse_rake_command}'"
     )
     logger.info("Enable plugins command: %s", full_command)
-    action = await unit.run(full_command)
-    await action.wait()
-    logger.info(action.results)
-    assert action.results["return-code"] == 0, "Enable plugins failed"
+    task = juju.exec(full_command, unit=app_name + "/0")
+    logger.info(task.results)
 
-    yield application
+    yield types.App(app_name)
```

Comment:
If the new type encapsulates a single name, why not use a string?

```
@pytest.fixture(scope="module"):
def app_name(...) -> str:
    ...
```

Or if the idea is that there may be more fields... maybe `return juju.status().apps[app_name]` ?

Reply:
I wondered about that too. However, I think it's better as a proper `App` type as an `app_name` fixture to me implies it's very lightweight and just going to return the application name. But this is heavy and deploys a bunch of stuff, so the name "app" and a special name `types.App` type seems to indicate that intent better.